src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -195,6 +199,24 @@ private ImmutableArray<ClrInfo> GetOrCreateClrVersions()
             return _clrs;
         }
 
+        /// <summary>
+        /// Cleans up the temporary directory and symlinks for the Linux DAC workaround.
+        /// </summary>
+        private void CleanupTempDirectory()
+        {
+            string? tempDirectory = Interlocked.Exchange(ref _tempDirectory, null);
```

Comment:
Why do we need `Interlocked.Exchange` here?  The user shouldn't call Dispose twice and it won't hurt if they do.  Using `Interlocked.Exchange` doesn't _hurt_ here but it makes the code a little harder to understand, as in someone walking up to the codebase might think this is part of some multithreaded locking scheme when it's not.

I'd normally recommend replacing this line with `if (_tempDirectory != null && Directory.Exists(_tempDirectory)` and never assigning `_tempDirectory` back to null.  It's ok if the user mistakenly calls Dispose, even from multiple threads, since the catch should protect us below.  (However, as mentioned above there's still a race in this overall design, so this is general feedback that I don't expect you to address before we figure out a design that doesn't race.)